pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/SegmentUploader.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.manager.realtime;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+
+public interface SegmentUploader {
+  // Returns the URI of the uploaded segment. null if the upload fails.
+  URI uploadSegment(File segmentFile)
+      throws URISyntaxException;
```

Comment:
Why does this throw URISyntax exception? We are providing a File handle, it does not seem correct to let the upper layer handle the URI syntax failure. The uploader, when constructed, should verify the URIs right?

Reply:
Removed the exception.